clang/lib/Driver/Driver.cpp

```diff
@@ -4886,6 +4886,9 @@ Action *Driver::BuildOffloadingActions(Compilation &C,
     // individually.
     for (Action *&A : DeviceActions) {
       if ((A->getType() != types::TY_Object &&
+           !(A->getOffloadingToolChain()->getTriple().getOS() ==
+                 llvm::Triple::OSType::AMDHSA &&
```

Comment:
me yelling into the clouds: maybe one day we can have the triple functions actually make sense for GPU targets